Branch-SDK/src/io/branch/referral/ServerRequest.java

```diff
@@ -279,5 +279,26 @@ protected boolean doesAppHasInternetPermission(Context context){
 		return result == PackageManager.PERMISSION_GRANTED;
 	}
 
+	/**
+	 * Updates the google ads parameters. This should be called only from a background thread since it involves GADS method invocation using reflection
+	 *
+	 * @param sysObserver {@link SystemObserver} instance.
+	 */
+	public void updateGAdsParams(SystemObserver sysObserver) {
+		try {
+			String advertisingId = sysObserver.getAdvertisingId();
+			if (advertisingId != null && getPost() != null) {
+				getPost().put(Defines.Jsonkey.GoogleAdvertisingID.getKey(), advertisingId);
```

Comment:
Here are the steps 
1.Create a server request [Main Thread]
2.Push to queue if there is no error and call precess request [Main thread] 
3.Request Fetched form queue [Main thread]
4.Add GAID  [Background Thread]
5.Process request [Background Thread]
6.Publish Result [UI thread]